### PR TITLE
Add histogram plot and update LM

### PR DESCRIPTION
**Pull request by jenna-tomkinson**
# Add histogram plot and update LM

*Update - changed to histogram over density plot

In this PR, I perform two analyses:

1. Add histogram plots using the number of cell neighbors per single-cell to compare the distributions between heart numbers and healthy versus failing cell type.
2. Update the LM plot using only non-treated healthy and failing cells to use cell neighbors instead of cell count per well as a co-variate.

### CODE REVIEW

File: 4.analyze_data/notebooks/density_plot/nbconverted/density_plot.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+# ## Generate density plot with only failing hearts
+
+# In[5]:
+
+
+# Filter out only failing hearts to generate plot
+filtered_df = annotated_df[annotated_df['Metadata_cell_type'] == 'Failing']
+
+sns.set_style("whitegrid")
+sns.kdeplot(
```
Comment (gwaybio) [inline 4.analyze_data/notebooks/density_plot/nbconverted/density_plot.py:108]: now that I think about it (and see the plots) a density plot might not make too much sense for a nominal variable (only whole numbers, 1,2,3,etc.). A histogram will work better TBH

The current plots and colors are effective and work well! I'd try a histogram instead to see if it improves.

**Review by gwaybio — APPROVED** [review]
One minor comment

Other discussion items:

- Wow those beta coefficients are HUGE for failing hearts now. Why is that? I would not expect to see them so large... I am suspect since the UMAPs didn't capture these variables, even though the LM is saying that healthy/failing has a big contribution.
- It seems that per cell neighbors is a better modeling variable... It is also very interesting to see a slight shift up in the failing cells.
- What do you think are the next steps of this analysis? How does this analysis influence our interpretations? Lastly, given this analysis, do we do anything prior to data splitting?

File: 4.analyze_data/notebooks/density_plot/nbconverted/density_plot.py (hunk 1 above)
Comment (jenna-tomkinson) [reply 4.analyze_data/notebooks/density_plot/nbconverted/density_plot.py]: Done! It is better to see single-cells count instead of density. It tells a similar story, but we get more info. I have implemented this change.

**Comment by jenna-tomkinson** [comment]
Per the discussion items:

> Wow those beta coefficients are HUGE for failing hearts now. Why is that? I would not expect to see them so large... I am suspect since the UMAPs didn't capture these variables, even though the LM is saying that healthy/failing has a big contribution.

I had to include the per cell neighbor count from a different data frame since it was removed during feature selection. It seems that pycytominer found it to be redundant. Should we be looking at all of the normalized features and not the feature selected ones and see how that impacts the LM?

> It seems that per cell neighbors is a better modeling variable... It is also very interesting to see a slight shift up in the failing cells.

I agree! It is also interesting that actin features seem to be lowered in terms of significance (not high beta-coefficients). Does it mean that this analysis isn't working properly since ground truth is that actin features are important for failing versus healthy? Or is my assumption "overfitting" in a way if that makes sense?

> What do you think are the next steps of this analysis? How does this analysis influence our interpretations? Lastly, given this analysis, do we do anything prior to data splitting?

This is a very good question. The number of cell neighbors contributes significantly to the features, so maybe removing all single-cells with any adjacent neighbors and seeing how that impacts the LM results and the r2 score. Maybe there would be more explained variance between failing versus healthy if all single cells were isolated.

I want to add here that this is what CellProfiler states when you are measuring `adjacent` cells: "In this mode, two objects must have adjacent boundary pixels to be neighbors."

There are two paths I see here to answer the question: "Do we do anything prior to data splitting?" and "How does this analysis influence our interpretations?"

1. Based on the fact that any value above 0 means that a single-cell is directly touching another single-cell, then we would be able to filter out only `isolated cells`. But based on the histogram, we would be removing a ton of data. But since we know that neighbors are greatly contributing to the features, then this might be a willing sacrifice. 
2. Just do not worry about the cell neighbors and keep all of the single-cells since it would be a lot of data to work with.

The next step I think would be to start on the structure of the machine learning model and set up splitting the data. We should have a conversation about this before training any model.

**Comment by jenna-tomkinson** [comment]
@gwaybio I will be merging now, thank you @gwaybio for reviewing since this is not the typical procedure 😄
